design/cross-cutting-concepts.qmd

```diff
@@ -0,0 +1,66 @@
+# Cross-cutting concepts
+
+## Privacy
+
+The Seedcase Product itself does not contain any data that requires
+considering privacy concerns. However, Seedcase will be used to manage
+data and has two main privacy concerns: privacy related to the data that
+is entered into the Data Resource; and privacy with the people working
+on a subset of the data from the Data Resource handled by Seedcase (the
+data project).
+
+The Seedcase Product will allow setting up multiple layers of user
+access and authorization levels to the Data Resource instance. As
+default there will be an administrator who will have access to
+everything to do with user roles and permissions. In principle it should
+be possible for that user to set Seedcase up in a way that they
+themselves cannot view any of the uploaded data, but this will not be a
+standard setup.
+
+The database back end will hold all information about users and their
+permissions. This is done by allowing the administrator to assign an
+individual user to either roles or groups, some of which will come
+pre-defined. It should also be possible for an administrator to create
+their own roles and groups based on the people working on a particular
+project that is using the Seedcase Product to handle their data.
+
+The Seedcase Product will ship with the following user roles pre-defined
+to make it easier for the data project administrator to ensure the
+security and privacy of the data entered into the system.
+
+-   **Administrator**: Initial user, will as a standard be assigned
+    permissions to do everything in the database. Create, read, update,
+    and delete (CRUD) permissions on data, database objects, and users
+    (including assigning other users to Administrator role).
+
+-   **Administrator light (CRUD of data)**: A user assigned this role
+    will have full CRUD permission on data, but not on user permissions,
+    tables, or other database objects.
+
+-   **Views only**: Likely to be a way of connecting specific users with
+    specific views created on the data contained in the database. These
+    views would be created by someone with a type of administrator role.
+
+-   **Data create-update**: A user with permission to enter data, and to
+    view the data that they have entered in the past.
+
+-   **DBMS viewer (read of data and database structure)**: Role for data
```

Comment:
I can't completely remember our discussion. I remember we mentioned about some of these roles not being needed anymore, but which ones are they? @K-Beicher and @rqding